lib/vrl/stdlib/src/chunks.rs

```diff
@@ -0,0 +1,153 @@
+}
+
+fn is_root_utf8_byte(byte: &u8) -> bool {
+    byte >> 6 != 0b10
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct Chunks;
+
+impl Function for Chunks {
+    fn identifier(&self) -> &'static str {
+        "chunks"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                kind: kind::BYTES,
+                required: true,
+            },
+            Parameter {
+                keyword: "chunk_size",
+                kind: kind::INTEGER,
+                required: true,
+            },
+        ]
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[Example {
+            title: "chunks by byte",
+            source: r#"chunks("foobar", 1)"#,
+            result: Ok("hi"),
+        }]
+    }
+
+    fn compile(
+        &self,
+        _state: (&mut state::LocalEnv, &mut state::ExternalEnv),
+        _ctx: &mut FunctionCompileContext,
+        mut arguments: ArgumentList,
+    ) -> Compiled {
+        let value = arguments.required("value");
+        let chunk_size = arguments.required("chunk_size");
+
+        Ok(Box::new(ChunksFn { value, chunk_size }))
+    }
+}
+
+#[derive(Debug, Clone)]
+struct ChunksFn {
+    value: Box<dyn Expression>,
+    chunk_size: Box<dyn Expression>,
+}
+
+impl Expression for ChunksFn {
+    fn resolve(&self, ctx: &mut Context) -> Resolved {
+        let value = self.value.resolve(ctx)?;
+        let chunk_size = self.chunk_size.resolve(ctx)?;
+
+        chunks(value, chunk_size)
+    }
+
+    fn type_def(&self, _state: (&state::LocalEnv, &state::ExternalEnv)) -> TypeDef {
+        TypeDef::array(Collection::from_unknown(Kind::bytes())).infallible()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    test_function![
+        chunks => Chunks;
+
+        unicode {
+            args: func_args![value: "你好",
+                             chunk_size: 4
+            ],
+            want: Ok(value!(["你", "好"])),
+            tdef: TypeDef::array(Collection::from_unknown(Kind::bytes())),
+        }
+
+        minimum_chunk_size {
+            args: func_args![value: "",
+            chunk_size: 2
+            ],
+            want: Err(r#""chunk_size" must be greater than or equal to 4 bytes"#),
+            tdef: TypeDef::array(Collection::from_unknown(Kind::bytes())),
+        }
+
+        mixed_ascii_unicode {
+            args: func_args![value: "ab你好",
+                             chunk_size: 4
+            ],
+            want: Ok(value!(["ab", "你", "好"])),
+            tdef: TypeDef::array(Collection::from_unknown(Kind::bytes())),
+        }
+
+        invalid_utf8 {
+            args: func_args![value: b"\xa0\xa1\xa2\xa3\xa4\xa5\xa6\xa7\xa8\xa9\xaa\xab\xac\xad\xae\xaf\xb0",
+                             chunk_size: 8
+            ],
+            want: Err("Bytes are not valid UTF-8"),
+            tdef: TypeDef::array(Collection::from_unknown(Kind::bytes())),
+        }
```

Comment:
This test macro is cool, but I think I need to learn more about the `tdef` parameter and valid values for it.

Reply:
`tdef` is the type definition that this function resolves to. In VRL, all expressions resolve to a value, and the type definition describes the possible values that it can resolve to, as well as whether it is fallible or not..

In this case,  `TypeDef::array(Collection::from_unknown(Kind::bytes()))` is just saying the function returns an array of _any_ size, where the elements must be `bytes`.